src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -124,17 +162,162 @@ export class MarkdownHoverParticipant implements IEditorHoverParticipant<Markdow
+		return disposables;
+	}
+
+	public async updateFocusedMarkdownHoverVerbosityLevel(action: HoverVerbosityAction): Promise<void> {
+		const model = this._editor.getModel();
+		if (
+			!this._focusInfo
+			|| !this._position
+			|| !this._context
+			|| !this._context.disposables
+			|| !model
+		) {
+			return;
+		}
+		const focusedIndex = this._focusInfo.index;
+		const currentHoverData = this._hoverData[focusedIndex];
+		if (!(currentHoverData instanceof VerboseMarkdownHover)) {
+			return;
+		}
+		const provider = currentHoverData.sourceProvider;
+		const hover = currentHoverData.hover;
+		this._verbosityLevelsByHoverIdx[focusedIndex] += (action === HoverVerbosityAction.Increase ? 1 : -1);
+		const context: HoverContext = { action, hover };
+		let newHover: Hover | null | undefined;
+		try {
+			newHover = await Promise.resolve(provider.provideHover(model, this._position, CancellationToken.None, context));
+		} catch (e) {
+			onUnexpectedExternalError(e);
+		}
+		if (!newHover) {
+			return;
+		}
+		this._hoverData[focusedIndex] = new VerboseMarkdownHover(newHover, provider, this, currentHoverData.range, false, currentHoverData.ordinal);
+		const renderedMarkdown = this._renderMarkdownHoversAndActions(
+			newHover.contents,
+			focusedIndex,
+			newHover.canIncreaseVerbosity,
+			newHover.canDecreaseVerbosity,
+			this._context.disposables
+		);
+		this._focusInfo.focusRemains = true;
+		this._focusInfo.element.replaceWith(renderedMarkdown);
+		this._focusInfo.element = renderedMarkdown;
+		this._context.onContentsChanged();
+		renderedMarkdown.focus();
+	}
+
+	private _renderMarkdownHoversAndActions(
+		hoverContents: IMarkdownString[],
+		hoverIndex: number,
+		canIncreaseVerbosity: boolean | undefined,
+		canDecreaseVerbosity: boolean | undefined,
+		store: DisposableStore,
+	): HTMLElement {
+
+		const contentsWrapper = $('div.hover-row');
+		contentsWrapper.tabIndex = 0;
+		const contents = $('div.hover-row-contents');
+		contentsWrapper.appendChild(contents);
+		renderMarkdownInContainer(
+			this._editor,
+			this._context,
+			contents,
+			hoverContents,
+			this._languageService,
+			this._openerService,
+			store
+		);
+		if (
+			(!canIncreaseVerbosity && !canDecreaseVerbosity)
+			|| !this._context
+			|| !this._context.disposables
+			|| !this._verbosityLevelsByHoverIdx
+		) {
+			return contentsWrapper;
+		}
+		const actionsContainer = $('div.verbosity-actions');
+		contentsWrapper.prepend(actionsContainer);
+
+		this._renderHoverExpansionAction(actionsContainer, HoverVerbosityAction.Increase, canIncreaseVerbosity ?? false, store);
+		this._renderHoverExpansionAction(actionsContainer, HoverVerbosityAction.Decrease, canDecreaseVerbosity ?? false, store);
+
+		const focusTracker = this._context.disposables.add(dom.trackFocus(contentsWrapper));
+		this._context.disposables.add(focusTracker.onDidFocus(() => {
+			this._focusInfo = {
+				index: hoverIndex,
+				element: contentsWrapper,
+				focusRemains: true
+			};
+		}));
```

Comment:
You sure you don't want to use `store` to track the disposable?

Reply:
I just noticed, it is the same DisposableStore instance. This should be cleaned up.